src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/SetAzureKeyVaultManagedStorageAccountAttribute.cs

```diff
@@ -0,0 +1,107 @@
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.Set, CmdletNoun.AzureKeyVaultManagedStorageAccountAttribute,
+        SupportsShouldProcess = true,
+        HelpUri = Constants.KeyVaultHelpUri )]
+    [OutputType( typeof( ManagedStorageAccount ) )]
+    public class SetAzureKeyVaultManagedStorageAccountAttribute : KeyVaultCmdletBase
+    {
+        #region Input Parameter Definitions
+
+        [Parameter( Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment." )]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Key Vault managed storage account name. Cmdlet constructs the FQDN of a managed storage account name from vault name, currently " +
+                          "selected environment and manged storage account name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountName )]
+        public string AccountName { get; set; }
+
+        [Parameter( Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Active key name. If not specified, the existing value of managed storage account's active key name remains unchanged" )]
+        [ValidateNotNullOrEmpty]
+        public string ActiveKeyName { get; set; }
+
+        [Parameter( Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Auto regenerate key. If not specified, the existing value of auto regenerate key of managed storage account remains unchanged" )]
+        [ValidateNotNull]
+        public bool? AutoRegenerateKey { get; set; }
+
+        [Parameter( Mandatory = false,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Regeneration period. If auto regenerate key is enabled, this value specifies the timespan after which managed storage account's inactive key" +
+                          "gets auto regenerated and becomes the active key. If not specified, the existing value of regeneration period of keys of managed storage " +
+                          "account remains unchanged" )]
+        [ValidateNotNull]
+        public TimeSpan? RegenerationPeriod { get; set; }
+
+        [Parameter( Mandatory = false,
+            HelpMessage = "If present, enables a use of a managed storage account key for sas token generation if value is true. Disables use of a managed " +
+                          "storage account key for sas token generation if value is false. If not specified, the existing value of the storage account's " +
+                          "enabled/disabled state remains unchanged." )]
+        [ValidateNotNull]
+        public bool? Enable { get; set; }
```

Comment:
This is along the design of existing key vault cmdlet of SetAzureKeyVaultKeyAttribute and SetAzureKeyVaultSecretAttribute

---
In reply to: [119727653](https://github.com/Azure/azure-powershell/pull/4061#discussion_r119727653) [](ancestors = 119727653)